apps/showcase/src/services/webcontainer/webcontainer-runner.ts

```diff
@@ -0,0 +1,275 @@
+import {FileSystemTree, IFSWatcher, WebContainer, WebContainerProcess} from '@webcontainer/api';
+import {Terminal} from '@xterm/xterm';
+import {
+  BehaviorSubject,
+  combineLatestWith,
+  distinctUntilChanged,
+  filter,
+  from,
+  map,
+  Observable,
+  switchMap
+} from 'rxjs';
+
+const createTerminalStream = (terminal: Terminal, cb?: (data: string) => void | Promise<void>) => new WritableStream({
+  write: (data) => {
+    if (cb) {
+      void cb(data);
+    }
+    terminal.write(data);
+  }
+});
+
+const makeProcessWritable = (process: WebContainerProcess, terminal: Terminal) => {
+  const input = process.input.getWriter();
+  terminal.onData((data) => input.write(data));
+  return input;
+};
+
+export class WebContainerRunner {
+  /**
+   * WebContainer instance which is available after the
+   */
+  public readonly instance: Promise<WebContainer>;
+  private readonly commands = new BehaviorSubject<{queue: string[]; cwd: string}>({queue: [], cwd: ''});
+  private readonly commandOnRun: Observable<{command: string; cwd: string} | undefined> = this.commands.pipe(
```

Comment:
By convention we add `$` as suffix for `Observable`
```suggestion
  private readonly commandOnRun$: Observable<{command: string; cwd: string} | undefined> = this.commands.pipe(
```